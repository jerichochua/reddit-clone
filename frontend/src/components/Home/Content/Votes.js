import React from 'react';

export default function Votes(props) {
  return (
    <div className='votes'>
      <button className='upvote-btn'></button>
      <span>{props.votes}</span>
      <button className='downvote-btn'></button>
    </div>
  );
}
