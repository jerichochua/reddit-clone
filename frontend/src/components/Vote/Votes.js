import React, { useState } from 'react';
import './Vote.css';

const Votes = ({ score }) => {
  const [isUpvoted, setIsUpvoted] = useState(false);
  const [isDownvoted, setIsDownvoted] = useState(false);

  const handleUpvote = () => {
    if (isUpvoted) {
      setIsUpvoted(false);
    } else if (isDownvoted) {
      setIsDownvoted(false);
      setIsUpvoted(true);
    } else {
      setIsUpvoted(true);
    }
  };

  const handleDownvote = () => {
    if (isDownvoted) {
      setIsDownvoted(false);
    } else if (isUpvoted) {
      setIsUpvoted(false);
      setIsDownvoted(true);
    } else {
      setIsDownvoted(true);
    }
  };

  return (
    <div className='votes-container'>
      <button
        className={`upvote-btn${isUpvoted ? ' voted' : ''}`}
        onClick={handleUpvote}
      ></button>
      <span>{score}</span>
      <button
        className={`downvote-btn${isDownvoted ? ' voted' : ''}`}
        onClick={handleDownvote}
      ></button>
    </div>
  );
};

export default Votes;
