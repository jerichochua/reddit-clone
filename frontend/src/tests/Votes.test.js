import renderer from 'react-test-renderer';
import Votes from '../components/Vote/Votes';

it('renders correctly', () => {
  const component = renderer.create(<Votes />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with score', () => {
  const component = renderer.create(<Votes score={1} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with upvote and downvote buttons', () => {
  const component = renderer.create(<Votes score={1} />);
  const tree = component.toJSON();
  expect(tree.children[0].props.className).toBe('upvote-btn');
  expect(tree.children[2].props.className).toBe('downvote-btn');
});
