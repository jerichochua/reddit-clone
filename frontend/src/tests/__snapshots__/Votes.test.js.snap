// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="votes-container"
>
  <button
    className="upvote-btn"
    onClick={[Function]}
  />
  <span />
  <button
    className="downvote-btn"
    onClick={[Function]}
  />
</div>
`;

exports[`renders correctly with score 1`] = `
<div
  className="votes-container"
>
  <button
    className="upvote-btn"
    onClick={[Function]}
  />
  <span>
    1
  </span>
  <button
    className="downvote-btn"
    onClick={[Function]}
  />
</div>
`;
